File: SinNombre/ontologiaRedis.js
var redis = require('redis')
var config = require('./config.json')
var ontologia = require('./ontologia.json')
var express=require('express') //Import 
var app=express();

function sendRedis(callback){	
	var redisClient = redis.createClient(config.redis.port, config.redis.host)
	console.log("Conectandose a redis")
	redisClient.on('error',function(error){
		console.log("Se presentó un error con redis")
	})
	console.log("Se envío mensaje a redis")
	callback(redisClient)
	console.log("Se envio acabó el mensaje")
	
}


function testRedis(redisClient){
	//redisClient.set('test','It's working,redis.print)
	redisClient.get('test',function(error,value){
			if(error){
				throw error
			}
			console.log("Testing a query in redis")
			console.log("-->" + value)
		})
	redisClient.quit();
}

function addSet(root, json){
	sendRedis(function(redisClient){
		for(var subRoot in json){
			clave = root + ':' + subRoot
			for(var i=0;i<json[subRoot].length;i++){
				console.log("sadd "+clave+" " +json[subRoot][i])
				sendRedis(function(redisClient){
					redisClient.sadd(clave,json[subRoot][i], function(err,res){
						if(err){
							console.log("Falló la ejecución del comando: " + err)
						}
					})
				})
				
			}
		}
	})
}


app.get('/', function(request, response){ //Start the main page 
	console.log("Conecting to Node Server...")
	//
	response.render('index.html');
	console.log("Connection completed");
	sendRedis(testRedis);
	addSet("nodos", ontologia.nodos);		
		addSet("relaciones", ontologia.relaciones);
		addSet("sinonimos", ontologia.sinonimos);
		
}).listen(3000)

File: SinNombre/ontologiaRedis.py
# -*- coding: utf-8 -*-
import json 
import redis
import Charmer
from FileManager import FileManager

fileManager = FileManager()
ontologia = fileManager.leerJson("ontologia.json")

def sendRedis(callback):
	redisClient = redis.StrictRedis(host='localhost', port=6379, db=0) 	
	#print("Conectandose a redis")
	#print("Se envío mensaje a redis")
	return callback(redisClient)	

def testRedis(redisClient):
	print(redisClient.get('test'))

def addSet(root, json):
	for subRoot in json:
		clave = root + ':' + subRoot
		for i in range(0,len(json[subRoot])):				
			print("sadd "+clave.encode("utf-8")+" " +json[subRoot][i].encode("utf-8"))
			def function(redisClient):
				return "Registros insertados: " + str(redisClient.sadd(clave,json[subRoot][i].encode("utf-8")))
			print(sendRedis(function))
def set(root, json):
	for subRoot in json:
		clave = root + ':' + subRoot
		for i in range(0,len(json[subRoot])):				
			print("set "+clave.encode("utf-8")+" " +json[subRoot][i].encode("utf-8"))
			def function(redisClient):
				return "Registros insertados: " + str(redisClient.set(clave,json[subRoot][i].encode("utf-8")))
			print(sendRedis(function))

def getSet(query):
	def function(redisClient):
		return redisClient.smembers(query)
	return sendRedis(function)

def addString(root,json):
        for subRoot in json:
                clave=subRoot
                print("set "+clave.encode("utf-8")+" " +json[subRoot].encode("utf-8"))
                def function(redisClient):
                        return "Registros insertados: " + str(redisClient.set(clave,json[subRoot].encode("utf-8")))
                print(sendRedis(function))
def cleanOnotology():
	def function(redisClient):
		redisClient.flushall()
	sendRedis(function)
def cargarOntologia():
	print cleanOnotology()
	addSet("nodes", ontologia["nodes"]);		
	addSet("relations", ontologia["relations"]);
	addString("synonymous", ontologia["synonymous"]);
	set("synonymous_atr", ontologia["synonymous_atr"]);

cargarOntologia()